qiskit/pulse/configuration.py

```diff
@@ -14,11 +14,28 @@
 Configurations for pulse experiments.
 """
 from typing import Dict, Union, Tuple, Optional
+import numpy as np
 
 from .channels import PulseChannel, DriveChannel, MeasureChannel
 from .exceptions import PulseError
 
 
+def _assert_nested_dict_equal(a, b):
+    for key in a:
```

Comment:
We need to ensure b does not have extra keys. Here is one way

```suggestion
def _assert_nested_dict_equal(a, b):
    if len(a) != len(b):
        return False
    for key in a:
```

Reply:
Thank you!